app/models/user.rb

```diff
@@ -463,6 +463,8 @@ def remove_from_algolia_index
   end
 
   def unsubscribe_from_newsletters
+    return if email.blank?
```

Comment:
Unsubscribe should be save from wherever we call it, so I added the check here, not in the service object.